pyomo/environ/__init__.py

```diff
@@ -106,4 +106,5 @@ def _import_packages():
     TerminationCondition, SolverStatus,
     )
 from pyomo.core.base.units_container import units
+from pyomo.core.expr.differentiate.differentiate import differentiate, DiffModes
```

Comment:
What concerns to folks have around "polluting" the `pyomo.environ` namespace?  What about:
```python
import pyomo.core.expr.calculus as calculus
```
Then users would 
```python
pyo.calculus.differentiate(my_expr, wrt=my_var, mode=pyo.calculus.differentiate.Modes.sympy
```

Reply:
Maybe we should leave this out of environ for now. We can always add it later. I have added the differentiate function to ```pyomo.core.expr```, so imports should not be bad:

```
from pyomo.core.expr import differentiate
```